geopy/geocoders/geocodio.py

```diff
@@ -0,0 +1,243 @@
+import json
+from functools import partial
+from urllib.parse import urlencode
+
+from geopy.exc import ConfigurationError, GeocoderQueryError, GeocoderQuotaExceeded
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import logger
+
+__all__ = ("Geocodio", )
+
+
+class Geocodio(Geocoder):
+    """Geocoder using the Geocod.io API.
+
+    Documentation at:
+        https://www.geocod.io/docs/
+
+    Pricing details:
+        https://www.geocod.io/pricing/
+
+    """
+
+    domain = 'api.geocod.io'
+    geocode_path = '/v1.6/geocode'
+    reverse_path = '/v1.6/reverse'
+
+    def __init__(
+        self,
+        api_key=None,
+        *,
+        scheme=None,
+        timeout=DEFAULT_SENTINEL,
+        proxies=DEFAULT_SENTINEL,
+        user_agent=None,
+        ssl_context=DEFAULT_SENTINEL,
+        adapter_factory=None
+    ):
+        """
+        :param str api_key:
+            A valid Geocod.io API key. (https://dash.geocod.io/apikey/create)
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+        :param callable adapter_factory:
+            See :attr:`geopy.geocoders.options.default_adapter_factory`.
+        """
+        super().__init__(
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+            adapter_factory=adapter_factory,
+        )
+        if api_key is None:
+            raise ConfigurationError('Must provide an api_key.')
```

Comment:
api_key is the only authentication method here, so the `api_key` parameter should just be required (i.e. don't have any default), then this check would become redundant. Example: https://github.com/geopy/geopy/blob/2.1.0/geopy/geocoders/what3words.py#L29

Reply:
Fixed in dbc5bf377cf3d6c42b211447eb31013e6a1115dc